scripts/update-bcd.ts

```diff
@@ -679,6 +684,48 @@ const skipCurrentBeforeSupport = skip("currentBeforeSupport", ({
   }
 });
 
+export const hasSupportMatrixContradictions = (
+  versionMap: BrowserSupportMap,
+  simpleStatement?: SimpleSupportStatement,
```

Comment:
@ChrisC do you plan to address this? If so I'll hold off with review of the recent changes.

Reply:
> @ChrisC do you plan to address this? If so I'll hold off with review of the recent changes.

@foolip I did try to address this and was hoping that with some smart sorting of the statements and versions, this would be simple to implement. Unfortunately, that approach did not work reliably. 

Instead, I'm going to add another `provide...` method after we get the `defaultStatements` to infer supported and unsupported ranges to check each version against. That will be the most robust solution for checking version support against multiple default statements. 

I would prefer to do that in a separate PR if possible (since I think it will be quite a bit more code to add), but if you have a strong preference for keeping it in this PR, I can continue with that work here.

Reply:
That sounds great @ChrisC, and fine to do it in a separate PR since it's not as simple as I thought initially.

Reply:
This is more or less what I tried. However, take the array of statements you provided as an example ☝🏽. 
```
"support": {
  "chrome": [
    {
      "version_added": "30"
    },
    {
      "version_added": "10",
      "version_removed": "20"
    }
  ]
}
```
Let's say you're testing version `10` with a a test result of `hasSupport === true` against this set of support statements. This should not show any contradictions. However, when tested for support against each statement individually, it would return `false` when tested against the the statement of `version_added: 30` and then send out a false positive signal for the need to update the statements. (Note that it would return `true` against the other statement.)

I suppose you could collect all the comparisons when iterating through the statements, and assume that as long as there's no inconsistency between the individual statement comparisons, there there's no need to update them. But that seemed a bit brittle and non-specific to me. See another approach where we test against ranges more specifically in https://github.com/openwebdocs/mdn-bcd-collector/pull/967.

Reply:
I'll take a look at your new PR, but for now just want to say that I don't think we should test a version ("10" in your example) against a part of a support statement, only the full array as one test.

If there is a contradiction we do need to figure out how which part of a support statement to update, or if we should add/remove a part, but I consider that a problem of handling more cases correctly, not identifying when an update is needed and logging.